### DISCUSSION TITLE
Helmrepository setup using api token

### DISCUSSION THREAD

**Discussion by soumyabk** [fluxcd/flux2#3029] [Q&A]
We are using use Jfrog artifactory to store our helm charts and I am trying to setup the helmrepository object. Is there an option to use an API token of a service account for authentication. 
I tried using it with the https authentication and that only seems to work with username and password.

```---
apiVersion: v1
kind: Secret
metadata:
  name: artifactory
  namespace: flux-system
data:
  username: <base64-username>
  password: <base64-password>
---
apiVersion: source.toolkit.fluxcd.io/v1beta1
kind: HelmRepository
metadata:
  name: artifactory
  namespace: flux-system
spec:
  interval: 5m0s
  url: <repository-url>
  secretRef:
    name: artifactory`

**Comment by soumyabk** [comment]
@stefanprodan if you have any pointers on this, it would be really helpful.

**Reply by stefanprodan** [reply]
It should work by providing the token in the password field.

**Comment by kingdonb** [comment]
> an API token of a service account for authentication.

Is this API token a Kubernetes Service Account token? Or is this an [Access token from JFrog Artifactory](https://www.jfrog.com/confluence/display/JFROG/Access+Tokens)? The first cannot be used to authenticate with Artifactory, but the second should work.

Please provide more information about the errors you received!
